samples/embedded/main.c

```diff
@@ -0,0 +1,215 @@
+    iree_infinite_timeout()));
+  iree_hal_buffer_view_release(result);
+
+  return iree_ok_status();
+}
+
+iree_status_t run_sample(iree_string_view_t bytecode_module_path,
+                         iree_string_view_t driver_name) {
+  iree_status_t status = iree_ok_status();
+
+  //===-------------------------------------------------------------------===//
+  // Instance configuration (this should be shared across sessions).
+  fprintf(stdout, "Configuring IREE runtime instance and '%s' device\n",
+          driver_name.data);
+  iree_runtime_instance_options_t instance_options;
+  iree_runtime_instance_options_initialize(&instance_options);
+  iree_runtime_instance_options_use_all_available_drivers(&instance_options);
+  iree_runtime_instance_t* instance = NULL;
+  if (iree_status_is_ok(status)) {
+    status = iree_runtime_instance_create(&instance_options,
+                                          iree_allocator_system(), &instance);
+  }
+  // TODO(#5724): move device selection into the compiled modules.
+  iree_hal_device_t* device = NULL;
+  if (iree_status_is_ok(status)) {
+    status = iree_runtime_instance_try_create_default_device(
+        instance, driver_name, &device);
+  }
+  //===-------------------------------------------------------------------===//
+
+  //===-------------------------------------------------------------------===//
+  // Session configuration (one per loaded module to hold module state).
+  fprintf(stdout, "Creating IREE runtime session\n");
+  iree_runtime_session_options_t session_options;
+  iree_runtime_session_options_initialize(&session_options);
+  iree_runtime_session_t* session = NULL;
+  if (iree_status_is_ok(status)) {
+    status = iree_runtime_session_create_with_device(
+        instance, &session_options, device,
+        iree_runtime_instance_host_allocator(instance), &session);
+  }
+  iree_hal_device_release(device);
+
+  fprintf(stdout, "Loading bytecode module at '%s'\n",
+          bytecode_module_path.data);
+  if (iree_status_is_ok(status)) {
+    status = iree_runtime_session_append_bytecode_module_from_file(
+        session, bytecode_module_path.data);
+  }
+  //===-------------------------------------------------------------------===//
+
+  //===-------------------------------------------------------------------===//
+  fprintf(stdout, "Training...\n");
+
+  print_state();
+  for (int i = 0; i < 10; i++) {
+    train(session);
+    print_state();
+  }
+  //===-------------------------------------------------------------------===//
+
+  //===-------------------------------------------------------------------===//
+  // Cleanup.
+  iree_runtime_session_release(session);
+  iree_runtime_instance_release(instance);
+  //===-------------------------------------------------------------------===//
+
+  return status;
+}
+
+int main(int argc, char** argv) {
+  if (argc < 2) {
+    fprintf(
+        stderr,
+        "Usage: embedded </path/to/embedded.vmfb> [<driver_name>]\n");
```

Comment:
I think for the final demo we want to embed the vmfb file in the binary.

Reply:
Yeah agreed, will add that in the next iteration.